pkg/queue/sharedmain/main_test.go

```diff
@@ -189,3 +190,76 @@ func TestQueueTraceSpans(t *testing.T) {
 		})
 	}
 }
+
+type UnsupportedEnvConfig struct {
+	EnableHTTP2AutoDetection bool `split_words:"true"` // optional
+}
+
+func TestEnv(t *testing.T) {
+	envVars := []struct {
+		name  string
+		value string
+	}{{
+		"ENABLE_HTT_P2_AUTO_DETECTION",
+		"true",
+	}, {
+		"ENABLE_HTTP_AUTO_DETECTION",
+		"true",
+	}, {
+		"CONTAINER_CONCURRENCY",
+		"10",
+	}, {
+		"QUEUE_SERVING_PORT",
+		"8080",
+	}, {
+		"QUEUE_SERVING_TLS_PORT",
+		"443",
+	}, {
+		"REVISION_TIMEOUT_SECONDS",
+		"1000",
+	}, {
+		"USER_PORT",
+		"8081",
+	}, {
+		"SERVING_LOGGING_CONFIG",
+		"",
+	}, {
+		"SERVING_LOGGING_LEVEL",
+		"info",
+	}, {
+		"SERVING_NAMESPACE",
+		"knative-serving",
+	}, {
+		"SERVING_CONFIGURATION",
+		"",
+	}, {
+		"SERVING_REVISION",
+		"rev",
+	}, {
+		"SERVING_POD",
+		"pod",
+	}, {
+		"SERVING_POD_IP",
+		"1.1.1.1",
+	}}
+
+	for _, v := range envVars {
+		t.Setenv(v.name, v.value)
+	}
+	var env config
+	if err := envconfig.Process("", &env); err != nil {
+		t.Fatal("Got unexpected error processing env:", err)
+	}
+	if !env.EnableHTTPAutoDetection {
+		t.Fatal("Flag ENABLE_HTTP_AUTO_DETECTION should be set to true")
+	}
+	var uEnv UnsupportedEnvConfig
+	if err := envconfig.Process("", &uEnv); err != nil {
+		t.Fatal("Got unexpected error processing env:", err)
+	}
+	// Splitting should have been ENABLE_HTTP2_AUTO_DETECTION
+	// Keeping this here as a warning
+	if !uEnv.EnableHTTP2AutoDetection {
```

Comment:
This should have failed.

Reply:
Do you want to keep that (as its a bit weird to check for "ENABLE_HTT_P2_AUTO_DETECTION"). Maybe we could have a check that the envs do not contain a number or something?

Reply:
Like looping trough `defaultEnv` and making sure everything is parsable?

Reply:
I might add just a comment. This is more to demonstrate the problem.

Reply:
Hm is that needed? If we make sure no one can add/use an env variable with a number, we no longer need to demonstrate the problem :) 

I'm fine both ways, I was just confused with that test.

Reply:
The test shows what fails, not intending to keep it. The comment helps people know the bug. Btw the project with the bug seems too old.